units/commit_queue.go

```diff
@@ -649,12 +615,17 @@ func addMergeTaskAndVariant(patchDoc *patch.Patch, project *model.Project, proje
 	}
 
 	// Merge task depends on all the tasks already in the patch
+	status := ""
+	if source == commitqueue.SourcePullRequest {
+		status = model.AllStatuses
```

Comment:
Can you explain this? This makes it seem like the PR merge will happen regardless of whether the other tasks succeed or fail, which seems wrong.

Reply:
added a comment here and in a similar spot in the git.merge_pr command explaining why